BK Precision 9130B Instrument

dennisfeng2 opened the pull request:
Added the beginning of the new instrument: BK Precision 9130B

dennisfeng2 commented:
I've noticed that there are some failed tests/checks with this pull request, but after taking a look at the error messages, it's still unclear to me whether these issues are coming from bugs in the pull request itself. For example, some of the errors involve import errors of VXI11Adapter and HP34410A, which my code doesn't seem to touch at all. Could someone point out to me why these errors are occurring, or anything I can fix from my end, so we can make progress on this PR?

dennisfeng2 commented:
Okay, merged with latest master and added the RST documentation. Could you take a look?

cjermain commented on docs/api/instruments/bkprecision/bkprecision9130b.rst:
```diff
@@ -0,0 +1,9 @@
+#########################
```
The header sections (i.e. `####`) should match the title in length.
```
##################################
BK Precision 9130B DC Power Supply
##################################
```

dennisfeng2 replied:
done

cjermain commented on pymeasure/instruments/bkprecision/bkprecision9130b.py:
```diff
@@ -0,0 +1,66 @@
+class BKPrecision9130B(Instrument):
+    """ Represents the BK Precision 9130B DC Power Supply interface for interacting with the instrument. """
+
+    current = Instrument.control('MEASure:SCALar:CURRent:DC?', 'SOURce:CURRent:LEVel:IMMediate:AMPLitude %g',
```
Please update the style with the comments in #198.

dennisfeng2 replied:
done

cjermain commented on pymeasure/instruments/bkprecision/bkprecision9130b.py:
```diff
@@ -0,0 +1,66 @@
+
+    @voltage.setter
+    def voltage(self, level):
+        if self.channel == 3:
```
One way to shorten this `if` statement, while keeping readability, is to write it as:
```
range = [0, 5] if self.channel == 3 else [0, 30]
new_level = truncated_range(level, range)
```

dennisfeng2 replied:
done

dennisfeng2 commented:
Hi, apologies that this is a bit of an old PR, but I finally got around to addressing the comments, and I'd like to push this forward. Are there any more code changes you'd like me to make, before we can merge it?

dennisfeng2 commented:
Bumping this, since it's probably relatively easy to review and merge (b/c it already has been code reviewed once)

bilderbuchi commented:
Looking good, thank you! It's never too late, we are also sometimes slow to get around to stuff. :-)

dennischenfeng commented:
Nice, no worries! Thank you :)

dennisfeng2 commented:
(Oops, I meant to thank you through this account rather than my other github account above) 
Thanks for merging!
